From 99ce805e13c809383850123ade3236f7ab95a388 Mon Sep 17 00:00:00 2001
From: Alpha Lin <alpha.lin@rock-chips.com>
Date: Thu, 16 Jul 2020 18:46:15 +0800
Subject: [PATCH] rockchip/iep: remove devm_* de-initialization call

Get rid of devm_* de-initialization call in iep driver,
for devm de-initialization api aren't necessary called
in modules driver.

Signed-off-by: Alpha Lin <alpha.lin@rock-chips.com>
Change-Id: I20f063d1cf30d15087fbf36f57b2d07fe1eff200
---
 drivers/video/rockchip/iep/iep_drv.c | 19 -------------------
 1 file changed, 19 deletions(-)

diff --git a/drivers/video/rockchip/iep/iep_drv.c b/drivers/video/rockchip/iep/iep_drv.c
index dc7b145d457f..9325b68d733f 100644
--- a/drivers/video/rockchip/iep/iep_drv.c
+++ b/drivers/video/rockchip/iep/iep_drv.c
@@ -1110,12 +1110,6 @@ static int iep_drv_probe(struct platform_device *pdev)
 err_misc_register:
 	free_irq(data->irq0, pdev);
 err_irq:
-	if (res) {
-		if (data->iep_base) {
-			devm_ioremap_release(&pdev->dev, res);
-		}
-		devm_release_mem_region(&pdev->dev, res->start, resource_size(res));
-	}
 err_ioremap:
 	wake_lock_destroy(&data->wake_lock);
 #ifdef IEP_CLK_ENABLE
@@ -1127,7 +1121,6 @@ static int iep_drv_probe(struct platform_device *pdev)
 static int iep_drv_remove(struct platform_device *pdev)
 {
 	struct iep_drvdata *data = platform_get_drvdata(pdev);
-	struct resource *res;
 
 	iep_iommu_info_destroy(iep_service.iommu_info);
 	iep_service.iommu_info = NULL;
@@ -1136,20 +1129,8 @@ static int iep_drv_remove(struct platform_device *pdev)
 
 	misc_deregister(&(data->miscdev));
 	free_irq(data->irq0, &data->miscdev);
-	res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
-	devm_ioremap_release(&pdev->dev, res);
-	devm_release_mem_region(&pdev->dev, res->start, resource_size(res));
 
 #ifdef IEP_CLK_ENABLE
-	if (data->aclk_iep)
-		devm_clk_put(&pdev->dev, data->aclk_iep);
-
-	if (data->hclk_iep)
-		devm_clk_put(&pdev->dev, data->hclk_iep);
-
-	if (data->pd_iep)
-		devm_clk_put(&pdev->dev, data->pd_iep);
-
 	pm_runtime_disable(data->dev);
 #endif
 
-- 
2.35.3

